skops/card/_parser.py

```diff
@@ -0,0 +1,353 @@
+        # right parent or child node, we need to keep track of the level of the
+        # headers. This cannot be done on the level of the markdown mapping,
+        # since it is not aware of the tree structure.
+        level, _, _ = item["c"]
+        content = self.mapping(item)
+        return content, level
+
+    def _post_process(self, res: str) -> str:
+        # replace Latin1 space
+        res = res.replace("\xa0", " ")
+
+        # pandoc creates ☒ and ☐ for to do items but GitHub requires [x] and [ ]
+        # for an item to be considered a to do item
+        res = res.replace("- ☒", "- [x]").replace("- ☐", "- [ ]")
+        return res
+
+    def generate(self) -> Card:
+        """Generate the model card instance from the parsed card.
+
+        Returns
+        -------
+        card : :class:`~skops.card.Card`
+            The parsed model card instance. If not further modified, the output
+            of saving that card should be (almost) identical to the initial
+            model card.
+        """
+        section: Section | None = None
+        section_trace: list[str] = []
+        card = Card(None, template=None)
+
+        # Parsing the flat structure, not recursively as in pandocfilters. After
+        # visiting the parent node, it's not necessary to visit its child nodes,
+        # because the mapping class already takes care of visiting the child
+        # nodes.
+        for item in json.loads(self.source)["blocks"]:
+            if item["t"] == "Header":
+                content, level = self._parse_header(item, section_trace=section_trace)
+                res = self._post_process(content)
+                section_trace = section_trace[: level - 1] + [res]
+                section = self._add_section(res, card=card, section_trace=section_trace)
+            else:
+                res = self._post_process(self.mapping(item))
+                self._add_content(res, section=section)
+
+        return card
+
+
+def _get_pandoc_version() -> list[int]:
+    """Shell out to retrieve the pandoc version
+
+    Raises
+    ------
+    RuntimeError
+        If pandoc version could not be determined, raise a ``RuntimeError``.
+
+    Returns
+    -------
+    pandoc_version : list[int]
+        The pandoc version as a list of ints.
+    """
+    proc = subprocess.run(
+        ["pandoc", "--version"],
+        capture_output=True,
+    )
+    version_info = str(proc.stdout.decode("utf-8")).split("\n", 1)[0]
+    if not version_info.startswith("pandoc "):
+        raise RuntimeError("Could not determine version of pandoc")
+
+    _, _, actual_version = version_info.partition(" ")
+    pandoc_version = [int(v) for v in actual_version.split(".")]
+    return pandoc_version
+
+
+def _check_version_greater_equal(
+    version: Sequence[int], min_version: Sequence[int]
+) -> None:
+    """Very bad version comparison function to ensure that the first version is
```

Comment:
We had a similar discussion in skorch: https://github.com/skorch-dev/skorch/issues/887

We decided not to use `packaging` because we didn't want to have a dependency on it. It seems that in skops, it is available (probably a transient dependency), but it's not an explicit requirement. If you are okay with explicitly depending on `packaging`, I can make the change.

Reply:
setuptools depends on it, I don't think `packaging` is a heavy dependency, I'm happy to include it as an explicit dep.